CMake/ITKSetPython3Vars.cmake

```diff
@@ -5,8 +5,9 @@
 # If the cmake variable "PYTHON_DEVELOPMENT_REQUIRED" is set to ON
 # then the development environments are found.
 
-if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "3.12.0")
+if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "3.15.0")
   # Use of PythonInterp and PythonLibs is deprecated since cmake version 3.12.0
+  # Unstable behavior of FindPython between CMake 3.12 and 3.15, so including in this
   # Only use deprecated mechanisms for older versions of cmake
   set(Python_ADDITIONAL_VERSIONS 3.9 3.8 3.7 3.6 3.5)
```

Comment:
Is this the right place to drop support for 3.5? It reached its end of life.

Reply:
Oh.  So actually, the code I fixed for CMake > 3.15 didn't have any checks on Python version.  There's an if further down:
```
  if(PYTHON_VERSION_STRING VERSION_LESS 3.5)
    # if python version is less than 3.5, unset so that it appears that no python version is found.
    # to emulate the same behavior as find(Python3 ..) from cmake 3.12.0+
    unset(PYTHON_EXECUTABLE)
    unset(PYTHONINTERP_FOUND)
    unset(PYTHON_VERSION_STRING)
    unset(PYTHON_INCLUDE_DIRS)
```
I'd suggest moving that check to the end of the file, so it happens regardless of CMake version.  Also, print some sort of message, e.g, "Python (version > 3.5) not found."

Reply:
To be clear, I think both locations would need to be modified.

Reply:
3.5 support removal could be added in a different patch.